ui/component/commentsList/view.jsx

```diff
@@ -206,23 +215,48 @@ function CommentList(props: Props) {
     }
 
     const handleCommentScroll = debounce(() => {
-      if (hasDefaultExpansion && shouldFetchNextPage(page, topLevelTotalPages, window, document)) {
+      if (
+        hasDefaultExpansion &&
+        !isFetchingComments &&
+        canDisplayComments &&
+        readyToDisplayComments &&
+        moreBelow &&
+        shouldFetchNextPage(page, topLevelTotalPages, window, document)
+      ) {
         setPage(page + 1);
       }
     }, DEBOUNCE_SCROLL_HANDLER_MS);
 
-    if (!isFetchingComments && readyToDisplayComments && moreBelow && spinnerRef && spinnerRef.current) {
-      if (shouldFetchNextPage(page, topLevelTotalPages, window, document, 0)) {
-        setPage(page + 1);
-      } else {
-        window.addEventListener('scroll', handleCommentScroll);
-        return () => window.removeEventListener('scroll', handleCommentScroll);
-      }
-    }
-  }, [hasDefaultExpansion, isFetchingComments, moreBelow, page, readyToDisplayComments, topLevelTotalPages]);
+    window.addEventListener('scroll', handleCommentScroll);
+    return () => window.removeEventListener('scroll', handleCommentScroll);
+  }, [
+    canDisplayComments,
+    hasDefaultExpansion,
+    isFetchingComments,
+    moreBelow,
+    page,
+    readyToDisplayComments,
+    topLevelTotalPages,
+  ]);
+
+  // Wait to only display topLevelComments after resolved or else
+  // other components will try to resolve again, like channelThumbnail
+  useEffect(() => {
+    if (!isResolvingComments) setCommentsToDisplay(topLevelComments);
+  }, [isResolvingComments, topLevelComments]);
+
+  // Batch resolve comment channel urls
+  useEffect(() => {
+    if (alreadyResolved) return;
+
+    const urisToResolve = [];
+    topLevelComments.map(({ channel_url }) => channel_url !== undefined && urisToResolve.push(channel_url));
```

Comment:
`filter` doesn't work here because it needs only the `channel_url` instead of the whole `Comment` object, so it will still need a new array from its values

Reply:
Oops, missed that.  If that's the case, doing the `push` in a regular loop or  `forEach` would be more appropriate.
- `map` will generate an array based on the return of `push` (`[1, 2, 3, 4 ..]`) that gets discarded in this case -- wasted memory allocation.
- In terms of readability and intention, `map` tells someone that we are generating a same-sized array with transformed object, which is not the case here since we are doing some filtering (skipping undefined urls).

No need to change for now, just something to keep in mind.